syntax = "proto3";

service SymbolRegistry {
  rpc GetSymbolMetadata (SymbolRequest) returns (SymbolMetadata);
  rpc ListAllSymbols (Empty) returns (SymbolList); // ðŸ‘ˆ New
}

message SymbolRequest {
  string symbol = 1;
}

message SymbolMetadata {
  string symbol = 1;
  double tick_size = 2;
  int32 precision = 3;
  double max_leverage = 4;
  double initial_margin = 5;
  double maintenance_margin = 6;
}

// ðŸ‘‡ New message for list response
message SymbolList {
  repeated string symbols = 1;
}

message Empty {}
